lib/ClangImporter/ClangDerivedConformances.cpp

```diff
@@ -0,0 +1,97 @@
+//===--- ClangDerivedConformances.cpp -------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "ClangDerivedConformances.h"
+#include "swift/AST/NameLookup.h"
+
+using namespace swift;
+
+void swift::conformToCxxIteratorIfNeeded(
+    ClangImporter::Implementation &impl, StructDecl *decl,
```

Comment:
Would we ever want this to work for foreign reference types? 
```suggestion
    ClangImporter::Implementation &impl, NominalTypeDecl *decl,
```

Reply:
Yeap, you're right, thanks!